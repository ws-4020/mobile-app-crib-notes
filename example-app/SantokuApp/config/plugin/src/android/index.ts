/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export * from './withAndroidAddAppActivity';
export * from './withAndroidAddAppActivityAndroidManifest';
export * from './withAndroidAddLicensePlugin';
export * from './withAndroidAddManifestPlaceholders';
export * from './withAndroidAppBuildGradleForRelease';
export * from './withAndroidCopyTemplateJavaFile';
export * from './withAndroidDisabledWindowDrawsSystemBarBackgrounds';
export * from './withAndroidEnabledStatusBarTranslucent';
export * from './withAndroidCopyMainActivity';
export * from './withAndroidMoveDevSettingsActivityToDebugAndroidManifest';
export * from './withAndroidCopySplashScreenImages';
export * from './withAndroidCopyTemplateSplashScreenXml';
export * from './withAndroidFlexibleSplashScreen';
