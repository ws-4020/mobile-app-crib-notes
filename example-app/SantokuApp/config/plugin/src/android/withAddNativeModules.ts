import {ConfigPlugin, withPlugins} from '@expo/config-plugins';

import {withAddNativeModulePackages} from './withAddNativeModulePackages';
import {withCopyTemplateJavaFile} from './withCopyTemplateJavaFile';

const NATIVE_MODULE_DIR = './config/plugin/template/android/app/src/main/java/com/helloworld';
const THROW_ERROR_MODULE_FILE_NAME = 'ThrowErrorModule.java';
const THROW_ERROR_PACKAGE_FILE_NAME = 'ThrowErrorPackage.java';

/**
 * アプリで作成したNativeModulesを追加します。
 */
export const withAddNativeModules: ConfigPlugin = config => {
  return withPlugins(config, [
    [withCopyTemplateJavaFile, {srcDir: NATIVE_MODULE_DIR, fileName: THROW_ERROR_MODULE_FILE_NAME}],
    [withCopyTemplateJavaFile, {srcDir: NATIVE_MODULE_DIR, fileName: THROW_ERROR_PACKAGE_FILE_NAME}],
    withAddNativeModulePackages,
  ]);
};
