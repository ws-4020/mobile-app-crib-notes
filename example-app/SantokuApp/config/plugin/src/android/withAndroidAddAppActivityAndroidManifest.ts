/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {ConfigPlugin, withAndroidManifest} from 'expo/config-plugins';

/**
 * AppActivityをAndroidManifest.xmlに追加します。
 * @param config ExpoConfig
 */
export const withAndroidAddAppActivityAndroidManifest: ConfigPlugin = config => {
  return withAndroidManifest(config, config => {
    const androidManifest = config.modResults;
    const mainApplication = androidManifest.manifest.application?.find(a => a.$['android:name'] === '.MainApplication');
    if (!mainApplication) {
      // MainApplicationは必ず存在する想定
      throw new Error('MainApplication does not exist in AndroidManifest.');
    }
    const originalMainActivity = mainApplication.activity?.find(a => a.$['android:name'] === '.MainActivity');
    if (!originalMainActivity) {
      throw new Error('MainActivity does not exist in AndroidManifest application.');
    }
    const mainActivity = {
      ...originalMainActivity,
      $: {
        'android:name': '.MainActivity',
        'android:theme': '@style/Theme.App.SplashScreen',
        'android:exported': 'true' as const,
      },
    };
    const restApplication =
      androidManifest.manifest.application?.filter(a => a.$['android:name'] !== '.MainApplication') ?? [];
    const restActivity = mainApplication.activity?.filter(a => a.$['android:name'] !== '.MainActivity') ?? [];

    config.modResults = {
      ...androidManifest,
      manifest: {
        ...androidManifest.manifest,
        application: [
          {
            ...mainApplication,
            activity: [
              mainActivity,
              {
                $: {
                  ...originalMainActivity.$,
                  'android:name': '.AppActivity',
                  'android:exported': 'false',
                  'android:theme': undefined,
                },
              },
              ...restActivity,
            ],
          },
          ...restApplication,
        ],
      },
    };
    return config;
  });
};
