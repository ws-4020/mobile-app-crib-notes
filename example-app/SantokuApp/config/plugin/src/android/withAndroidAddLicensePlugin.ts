/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {ConfigPlugin, withAppBuildGradle} from 'expo/config-plugins';

/**
 * libraryのライセンス一覧を取得するpluginを追加します。
 * @param config ExpoConfig
 */
export const withAndroidAddLicensePlugin: ConfigPlugin = config => {
  return withAppBuildGradle(config, config => {
    if (config.modResults.language === 'groovy') {
      config.modResults.contents = apply(config.modResults.contents);
    } else {
      throw new Error('In this app, only groovy is supported as the language for build.gradle.');
    }
    return config;
  });
};

const apply = (buildGradle: string): string => {
  // plugins block は記述場所に制限がある。他の plugin 追加とコンフリクトしないようにするため既存の biuldGradle を挟み込むように追加する
  return `
plugins {
  id "com.jaredsburrows.license" version "0.9.3"
}
${buildGradle}

licenseReport {
  generateCsvReport  = false
  generateHtmlReport = false
  generateJsonReport = true  // android/app/build/reports/licenses/licenseReleaseReport.json
  generateTextReport = false
}
`.trimStart();
};
