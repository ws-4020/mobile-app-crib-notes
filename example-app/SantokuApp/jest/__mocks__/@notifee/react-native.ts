/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

export default mock;
