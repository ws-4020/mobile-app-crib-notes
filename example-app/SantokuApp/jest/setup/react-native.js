/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
