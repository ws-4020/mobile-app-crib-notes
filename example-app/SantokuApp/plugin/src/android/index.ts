export * from './withAddAppActivity';
export * from './withAddAppActivityAndroidManifest';
export * from './withAddReleaseSigningConfigBuildGradle';
export * from './withCopyTemplateJavaFile';
export * from './withRemoveUsesClearTextTraffic';
export * from './withDisabledWindowDrawsSystemBarBackgrounds';
export * from './withEnabledStatusBarTranslucent';
export * from './withCopyMainActivity';
export * from './withMoveDevSettingsActivityToDebugAndroidManifest';
