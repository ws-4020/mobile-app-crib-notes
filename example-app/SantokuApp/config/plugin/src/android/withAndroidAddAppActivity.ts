/**
 * Copyright 2023 TIS Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {ConfigPlugin, withPlugins} from 'expo/config-plugins';

import {withAndroidAddAppActivityAndroidManifest} from './withAndroidAddAppActivityAndroidManifest';
import {withAndroidCopyMainActivity} from './withAndroidCopyMainActivity';
import {withAndroidCopyTemplateJavaFile} from './withAndroidCopyTemplateJavaFile';
import {ANDROID} from '../constants';

const MAIN_ACTIVITY_FILE_NAME = 'MainActivity.java';
const APP_ACTIVITY_CLASS_NAME = 'AppActivity';

/**
 * Transition中にActivityのbackgroundが表示される問題があるため、Splash Screen用とReact Native用にActivityを分けます。
 * https://github.com/software-mansion/react-native-screens/issues/380
 *
 * 1. 「MainActivity.java」を「AppActivity.java」というファイル名でコピー（併せて、「AppActivity.java」のクラス名を「AppActivity」に変更）
 * 2. テンプレートのMainActivity.javaを「android/app/src/main/[パッケージ名]」にコピー（既存のMainActivity.javaを上書き）
 * 3. AndroidManifest.xmlにAppActivityを追加
 *
 * 1でMainActivity.javaを単純にリネームしないでコピーしているのは、MainActivity.javaが存在しないとExpo prebuild中にエラーが発生するためです。
 * @param config ExpoConfig
 */
export const withAndroidAddAppActivity: ConfigPlugin = config => {
  return withPlugins(config, [
    [
      withAndroidCopyTemplateJavaFile,
      {srcDir: ANDROID.PLUGIN_TEMPLATE_MAIN_ACTIVITY_DIR, fileName: MAIN_ACTIVITY_FILE_NAME},
    ],
    [withAndroidCopyMainActivity, {toClassName: APP_ACTIVITY_CLASS_NAME}],
    withAndroidAddAppActivityAndroidManifest,
  ]);
};
