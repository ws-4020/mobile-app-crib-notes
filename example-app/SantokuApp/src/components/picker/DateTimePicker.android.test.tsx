import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {StyleSheet, TextInputProps, View, ViewStyle} from 'react-native';

import {DateTimePicker} from './DateTimePicker.android';
import {DateTimePickerItemsAndroidProps} from './DateTimePickerItems';

describe('DateTimePicker only with required props', () => {
  it('renders successfully if invisible', () => {
    const sut = render(
      <DateTimePicker textInputProps={{testID: 'textInput'}} pickerItemsProps={{testID: 'pickerItems'}} />,
    );

    expect(sut).toMatchSnapshot('DateTimePicker if invisible.');

    const textInput = sut.queryByTestId('textInput');
    const pickerItems = sut.queryByTestId('pickerItems');
    expect(textInput).not.toBeNull();
    expect(pickerItems).toBeNull();
  });

  it('renders successfully if visible', () => {
    // selectedValueに固定値を入れないと、Snapshot取得時に毎回値が変わってしまうので、OptionalPropsですが指定しています。
    const selectedValue = new Date(1656601200000);
    // 自動テストだと、displayを指定しないとレンダリングされなかったため、OptionalPropsですがdisplayAndroidを指定してます。
    const displayAndroid = 'calendar';
    // formatTextを指定しないと、タイムゾーンの表記が環境によって違うので、どの環境でも変わらないように指定しています。
    const formatText = (date?: Date) => (date ? date.toISOString() : '');
    const sut = render(
      <DateTimePicker
        selectedValue={selectedValue}
        textInputProps={{testID: 'textInput'}}
        pickerItemsProps={{testID: 'pickerItems'}}
        displayAndroid={displayAndroid}
        formatText={formatText}
      />,
    );
    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);

    expect(sut).toMatchSnapshot('DateTimePicker if visible.');

    const textInput = sut.queryByTestId('textInput');
    const pickerItems = sut.queryByTestId('pickerItems');
    expect(textInput).not.toBeNull();
    expect(pickerItems).not.toBeNull();
  });
});

describe('DateTimePicker with default value', () => {
  it('defaultValue should be set at open if selectedValue does not exist,', () => {
    const defaultValue = new Date(1654786800000);
    const sut = render(
      <DateTimePicker
        defaultValue={defaultValue}
        pickerItemsProps={{testID: 'pickerItems'}}
        displayAndroid="calendar"
      />,
    );
    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);
    expect(sut.getByTestId('pickerItems').props.date).toBe(defaultValue.getTime());
  });
});

describe('DateTimePicker with all props', () => {
  it('should be applied all properly with default xxx component', () => {
    const onSelectedItemChange = jest.fn();
    const onDone = jest.fn();
    const formatText = jest.fn();
    const selectedValue = new Date(1643641200000);
    const maximumDate = new Date(1656601200000);
    const minimumDate = new Date(1612105200000);
    /**
     * onDoneは自動テストではイベント発火できませんでした
     */
    const sut = render(
      <DateTimePicker
        selectedValue={selectedValue}
        maximumDate={maximumDate}
        minimumDate={minimumDate}
        defaultValue={maximumDate}
        onSelectedItemChange={onSelectedItemChange}
        placeholder="please select..."
        textInputProps={{style: {color: 'red'}, testID: 'textInput'}}
        pickerItemsProps={{style: {backgroundColor: 'yellow'}, testID: 'pickerItems'}}
        onDone={onDone}
        formatText={formatText}
        mode="date"
        displayAndroid="calendar"
      />,
    );

    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);

    expect(sut).toMatchSnapshot('DateTimePicker all properly with default xxx component.');

    // assert PickerItems
    const pickerItems = sut.getByTestId('pickerItems');
    const pickerItemsProps = pickerItems.props as DateTimePickerItemsAndroidProps;
    // selectedValueはDateTimePickerItemsAndroidPropsで定義されていない「date」でレンダリングされます
    // @ts-ignore
    expect(pickerItemsProps.date).toBe(selectedValue.getTime());
    expect(pickerItemsProps.maximumDate).toBe(maximumDate.getTime());
    expect(pickerItemsProps.minimumDate).toBe(minimumDate.getTime());
    expect(pickerItemsProps.mode).toBe('date');
    // displayはdisplayIOSというPropでレンダリングされます
    // Platform.OSをモック化してandroidを返却するようにしても「displayIOS」でレンダリングされます
    // @ts-ignore
    expect(pickerItemsProps.displayIOS).toBe('calendar');
    const pickerItemsStyle = pickerItemsProps.style as ViewStyle;
    const flattenPickerItemsStyle = StyleSheet.flatten(pickerItemsStyle);
    expect(flattenPickerItemsStyle.backgroundColor).toBe('yellow');

    // assert textInput
    const textInput = sut.getByTestId('textInput');
    const textInputProps = textInput.props as TextInputProps;
    expect(textInputProps.style).toEqual({color: 'red'});
  });

  it('should be applied all properly with custom xxx component', () => {
    const CustomTextInput = <View testID="customTextInput" />;
    const onSelectedItemChange = jest.fn();
    const onDone = jest.fn();
    const formatText = jest.fn();
    const selectedValue = new Date(1643641200000);
    const maximumDate = new Date(1656601200000);
    const minimumDate = new Date(1612105200000);
    /**
     * onDoneは自動テストではイベント発火できませんでした
     */
    const sut = render(
      <DateTimePicker
        selectedValue={selectedValue}
        maximumDate={maximumDate}
        minimumDate={minimumDate}
        defaultValue={maximumDate}
        onSelectedItemChange={onSelectedItemChange}
        placeholder="please select..."
        textInputProps={{testID: 'defaultTextInput'}}
        pickerItemsProps={{style: {backgroundColor: 'yellow'}, testID: 'pickerItems'}}
        textInputComponent={CustomTextInput}
        onDone={onDone}
        formatText={formatText}
        mode="date"
        displayAndroid="calendar"
      />,
    );

    const pressableContainer = sut.getByTestId('pressableContainer');
    fireEvent.press(pressableContainer);

    expect(sut).toMatchSnapshot('DateTimePicker all properly with custom xxx component.');

    // assert PickerItems
    const pickerItems = sut.getByTestId('pickerItems');
    const pickerItemsProps = pickerItems.props as DateTimePickerItemsAndroidProps;
    // selectedValueはDateTimePickerItemsAndroidPropsで定義されていない「date」でレンダリングされます
    // @ts-ignore
    expect(pickerItemsProps.date).toBe(selectedValue.getTime());
    expect(pickerItemsProps.maximumDate).toBe(maximumDate.getTime());
    expect(pickerItemsProps.minimumDate).toBe(minimumDate.getTime());
    expect(pickerItemsProps.mode).toBe('date');
    // displayはdisplayIOSというPropでレンダリングされます
    // Platform.OSをモック化してandroidを返却するようにしても「displayIOS」でレンダリングされます
    // @ts-ignore
    expect(pickerItemsProps.displayIOS).toBe('calendar');
    const pickerItemsStyle = pickerItemsProps.style as ViewStyle;
    const flattenPickerItemsStyle = StyleSheet.flatten(pickerItemsStyle);
    expect(flattenPickerItemsStyle.backgroundColor).toBe('yellow');

    // assert textInput
    const defaultTextInput = sut.queryByTestId('defaultTextInput');
    const customTextInput = sut.queryByTestId('customTextInput');
    expect(defaultTextInput).toBeNull();
    expect(customTextInput).not.toBeNull();
  });
});
