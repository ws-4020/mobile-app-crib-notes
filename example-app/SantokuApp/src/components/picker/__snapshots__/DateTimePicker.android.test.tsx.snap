// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker only with required props renders successfully if invisible: DateTimePicker if invisible. 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="pressableContainer"
>
  <View
    pointerEvents="box-only"
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      rejectResponderTermination={true}
      testID="textInput"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`DateTimePicker only with required props renders successfully if visible: DateTimePicker if visible. 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableContainer"
  >
    <View
      pointerEvents="box-only"
    >
      <TextInput
        allowFontScaling={true}
        editable={false}
        rejectResponderTermination={true}
        testID="textInput"
        underlineColorAndroid="transparent"
        value="2022-05-30T00:00:00.000Z"
      />
    </View>
  </View>,
  <RNDateTimePicker
    date={1653868800000}
    displayIOS="calendar"
    enabled={true}
    mode="date"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 216,
      }
    }
    testID="pickerItems"
  />,
]
`;

exports[`DateTimePicker with all props should be applied all properly with custom xxx component: DateTimePicker all properly with custom xxx component. 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableContainer"
  >
    <View
      pointerEvents="box-only"
    >
      <View
        testID="customTextInput"
      />
    </View>
  </View>,
  <RNDateTimePicker
    date={1640995200000}
    displayIOS="calendar"
    enabled={true}
    maximumDate={1653955200000}
    minimumDate={1483228800000}
    mode="date"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 216,
        },
        Object {
          "backgroundColor": "yellow",
        },
      ]
    }
    testID="pickerItems"
  />,
]
`;

exports[`DateTimePicker with all props should be applied all properly with default xxx component: DateTimePicker all properly with default xxx component. 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="pressableContainer"
  >
    <View
      pointerEvents="box-only"
    >
      <TextInput
        allowFontScaling={true}
        editable={false}
        placeholder="please select..."
        rejectResponderTermination={true}
        style={
          Object {
            "color": "red",
          }
        }
        testID="textInput"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>,
  <RNDateTimePicker
    date={1640995200000}
    displayIOS="calendar"
    enabled={true}
    maximumDate={1653955200000}
    minimumDate={1483228800000}
    mode="date"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 216,
        },
        Object {
          "backgroundColor": "yellow",
        },
      ]
    }
    testID="pickerItems"
  />,
]
`;
